BootloaderCommonPkg/Library/BuildFdtLib/X86_BuildFdtLib.c

```diff
@@ -0,0 +1,909 @@
+/** @file
+  Copyright (c) 2024, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <PiPei.h>
+#include <IndustryStandard/Pci22.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/DebugLib.h>
+#include <Library/HobLib.h>
+#include <Library/PcdLib.h>
+#include <Library/IoLib.h>
+#include <Library/PrintLib.h>
+#include <Library/FdtLib.h>
+#include <Library/PciLib.h>
+#include <UniversalPayload/UniversalPayload.h>
+#include <UniversalPayload/AcpiTable.h>
+#include <UniversalPayload/SerialPortInfo.h>
+#include <UniversalPayload/PciRootBridges.h>
+#include <UniversalPayload/DeviceTree.h>
+#include <UniversalPayload/SmbiosTable.h>
+#include <Guid/GraphicsInfoHob.h>
+#include <Guid/UniversalPayloadBase.h>
+#include <Library/BootloaderCoreLib.h>
+
+#define IGD_BUS_NUM   0x00
+#define IGD_DEV_NUM   0x02
+#define IGD_FUN_NUM   0x00
+
+BOOLEAN               mResourceAssigned;
+
+CHAR8  *mMemoryAllocType[] = {
+  "Reserved",
+  "LoaderCode",
+  "LoaderData",
+  "boot-code",
+  "boot-data",
+  "runtime-code",
+  "runtime-data",
+  "ConventionalMemory",
+  "UnusableMemory",
+  "acpi",
+  "acpi-nvs",
+  "mmio",
+  "MemoryMappedIOPortSpace",
+  "PalCode",
+  "PersistentMemory",
+};
+
+/**
+  It will build FDT based on memory information from Hobs.
```

Comment:
Suggest blank line after function synopsis for readability and consistency. Same applies to other functions below.